Source/Core/Gen4/Encounters4.cpp

```diff
@@ -272,6 +272,125 @@ namespace Encounters4
                 }
             }
 
+            if (encounter == Encounter::SafariZone)
```

Comment:
Could you leave some comments as too what is happening here. I don't understand all the things happening. Especially with the block stuff